src/dbnode/storage/series/types.go

```diff
@@ -283,3 +329,26 @@ func NewStats(scope tally.Scope) Stats {
 func (s Stats) IncCreatedEncoders() {
 	s.encoderCreated.Inc(1)
 }
+
+// WriteType is an enum for warm/cold write types.
+type WriteType int
+
+const (
+	// UndefinedWriteType is an undefined write type.
+	UndefinedWriteType WriteType = iota
+
+	// WarmWrite represents warm writes (within the buffer past/future window).
+	WarmWrite
+
+	// ColdWrite represents cold writes (outside the buffer past/future window).
+	ColdWrite
+)
+
+// WriteOptions define different options for a write.
+type WriteOptions interface {
```

Comment:
Does this need to be constructed for each write or can we share one globally for all writes for a namespace? I'm just asking because we don't want to have to allocate a new interface for each write

Reply:
WriteOptions will be implemented by some concrete struct (we probably want to use pointer receiver)... there is no problem sharing the same opt per namespace. 

The important thing is that Options is immutable, so that it can be safely shared.

Reply:
Can we revert this to just be a struct?  

The options interfaces we have everywhere is when we want to share a single huge object with lots of long lived data structures (so they can all share the same ref to lots of fields).

These options though are passed down the stack and are not held on by any longly lived object so it's fine to keep as just a struct.